tasks/src/components/Results/Results.tsx

```diff
@@ -1,48 +1,54 @@
+import React, { useEffect } from 'react';
 import { Person } from 'src/pages/mainPage/MainPage.tsx';
 import { Card } from '../Card/Card.tsx';
 import { useNavigate } from 'react-router-dom';
+import style from 'src/components/form/Forms.module.scss';
+import { useDispatch, useSelector } from 'react-redux';
+import { AppDispatch, RootState, useGetQueryQuery } from 'src/store';
+import { setResultSlice } from 'src/store/resultSlice.ts';
+import { Loading } from '../Loading/Loading.tsx';
 
 interface ResultsProps {
-  query: string;
-  results: Person[];
-  isLoading: boolean;
-  setActiveCard: (card: Person) => void;
   isOpen: boolean;
   setIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
   activePage: string;
 }
 
-export const Results: React.FC<ResultsProps> = ({
-  results,
-  isLoading,
-  setActiveCard,
-  isOpen,
-  setIsOpen,
-  query,
-  activePage,
-}) => {
+export const Results: React.FC<ResultsProps> = ({ isOpen, setIsOpen, activePage }) => {
+  const query = useSelector((state: RootState) => state.input.input);
+  const queryResult = useSelector((state: RootState) => state.result.result);
```

Comment:
You can create a folder with selectors in the store folder, in which you can create selectors and then call this selector.
To do this, you only need to have one place where you can make some changes in the future, and it will spread to other places where you will call this selector.

Like this:
`export const getResult = (state: RootState) => state.result.result`

and call that function in components like this:
`const queryResult = useSelector(getResult);`